### PR TITLE
Fix compilation with primitive master

### PR DESCRIPTION
**Pull request by thomie**

### CODE REVIEW

**Comment by cartazio** [comment]
@thomie  this patch seems to fail on head and <=7.4

**Comment by thomie** [comment]
Thanks. I fixed the compilation with <= 7.4. The current `#ifdef` is hopefully future proof.

Any patch will fail with GHC HEAD though, because @hvr increased the version number of base to 4.9, but there is no release of `primitive` yet that allows base 4.9.

File: Data/Vector.hs [hunk 1]
```diff
@@ -161,7 +161,11 @@ module Data.Vector (
 
 import qualified Data.Vector.Generic as G
 import           Data.Vector.Mutable  ( MVector(..) )
+#if MIN_VERSION_primitive(0,6,2) || __GLASGOW_HASKELL__ >= 711
+import           Data.Primitive.Array hiding ( fromList, fromListN )
```
Comment (hvr) [inline Data/Vector.hs]: We can actually avoid the heavy CPP hammer...

Comment (thomie) [inline Data/Vector.hs]: Qualified import?

Comment (hvr) [inline Data/Vector.hs]: almost... qualified use =)

**Comment by thomie** [comment]
Thanks.

**Pull request closed without merging**